modules/contentbox/models/BaseEntityMethods.cfc

```diff
@@ -106,5 +119,36 @@ component mappedsuperclass="true"{
 
 		return result;
 	}
-	
+
+	/**
+	 * Returns the key (id field) of a given entity, either simple or composite keys.
+	 * If the key is a simple pk then it will return a string, if it is a composite key then it returns an array.
+	 * If the key cannot be identified then a blank string is returned.
+	 *
+	 * @entity The entity name or entity object
+	 *
+	 * @return string or array
+	 */
+	any function getKey(){
```

Comment:
@jclausen I know that this works great for convenience, but the reality that using this on every call for an entity is expensive. I prefer to NOT do lookups, but use the `this.pk` definition explicitly by each entity.

Reply:
I'm fine with that as long as it's dependable that the `this.pk` is present.  I assume that's why you didn't use ActiveEntity inheritance on any of these models.  I'll try to refactor it over to use that an eliminate the introspection